client/src/components/Upload/UploadModal.vue

```diff
@@ -262,12 +262,42 @@ const UploadModal = {
         currentFtp: function () {
             return this.currentUserId && this.ftpUploadSite;
         },
-
+        toData: function (items, history_id) {
+            var data = {
+                fetchRequest: null,
+                uploadRequest: null,
+            };
+            if (items && items.length > 0) {
+                var split = this.preprocess(items);
+                if (split.urls.length > 0) {
+                    data.fetchRequest = this.toFetchData(split.urls, history_id);
+                }
+                if (split.files.length > 0) {
+                    data.uploadRequest = this.toFileUploadData(split.files, history_id);
+                }
+            }
+            return data;
+        },
+        preprocess: function (items) {
+            var data = {
+                urls: [],
+                files: [],
+            };
+            for (var index in items) {
+                var it = items[index];
+                if (it.get("file_mode") != "new" || !it.get("url_paste").startsWith("http")) {
```

Comment:
Can we synchronize this with the backend - we do a bit more aggressive checks for URIs:

https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/parameters/grouping.py#L359
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/parameters/grouping.py#L25